### PR TITLE
Create GETTING-STARTED.md

### PR DESCRIPTION
**Pull request by ysangkok**
I have been searching for tools that allow for testable documentation, and this is what I came up with so far. It's finicky (especially the list-bin part), but submitting it here for review in case anybody has suggestions on how to improve it.

### CODE REVIEW

**Comment by qxjit** [comment]
@ysangkok This is great, thank you! Just a couple things I'd like to have before merging this

1) Can we make the getting started example not use any language extensions please?
2) Can we get a dockerized setup of `mdsh` that is run as port of `scripts/build.sh` and ci to ensure that we keep this example working

File: GETTING-STARTED.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+cat <<EOF > readme-test.t
+$ cd \$TESTDIR
+$ cp \$(cabal list-bin exe:\$(basename \$PWD) | tail -n1) \$OLDPWD
+$ cd \$OLDPWD
+$ ./orville-demo-*
+Right 2
+EOF
+```
+
+It can be verified to match using `prysk` (which can be installed using `pip3`):
```
Comment (telser) [inline GETTING-STARTED.md:78]: Maybe we should have a beginning section describing that this is based on Debian (and a version?) along with installing prerequisites and noting what they are for? That way we could level set at the very beginning. 

I didn't know of `prysk`, but it looks interesting.

Comment (ysangkok) [reply GETTING-STARTED.md:87]: It's just a forked https://bitheap.org/cram/

Comment (ysangkok) [reply GETTING-STARTED.md:87]: I added the Docker setup David suggested, does that take care of your concern?

**Comment by ysangkok** [comment]
@qxjit Added a Docker setup as suggested

**Comment by qxjit** [comment]
@ysangkok @telser  I tried the new `build.sh` locally and it doesn't seem to be building the `orville-docs` image for some reason. I think something is wrong with the `SKIP_IMAGE_BUILD` conditional or variable. It doesn't do a docker build whether I pass `--skip-image-build` or not.

If I remove the conditional wrapper then the `cd ../../docs` fails. It looks like it has one too many `..`s in it.

**Comment by ysangkok** [comment]
Where is `build.sh` supposed to be called from? Should it be agnostic to the working directory? I call it from the directory where it is, but it looks like you'd expect it to work when invoked from a different directory?

**Comment by qxjit** [comment]
@ysangkok We've been calling it from the `orville-postgresql-libpq` by using `./scripts/build.sh`.

**Comment by ysangkok** [comment]
Ok, I changed to support that. I avoided trying to work from any directory since it seems only possible with bashisms: https://stackoverflow.com/a/246128/309483

**Comment by qxjit** [comment]
@ysangkok I fixed the issues with the build script not building the docker images. This is good to merge with that commit included

**Review by qxjit — APPROVED** [review]

**Pull request merged**